### ISSUE TITLE
ZEVENT_VDEV_PATH, ZEVENT_VDEV_SPARE_PATHS contain partition names in vdev paths

### ISSUE THREAD

**Issue by bprotopopov** [zfsonlinux/zfs#2295]
I am looking at the automatic replacement of vdevs with spares on checksum errors. What I am observing is that the faulty device is degraded, but is not being replaced with a spare. 

Looking a bit further, instrumented checksum-spare.sh to log the env vars passed by zed, and the various steps in the script, getting this output: 
...
checksum-spare: Environment vars start
checksum-spare: pool test_pool
checksum-spare: class ereport.fs.zfs.checksum
checksum-spare: subclass checksum
checksum-spare: vdev path /dev/sdab1
checksum-spare: vdev guid 0x88100D7035C37C5C
checksum-spare: vdev spare paths /dev/sdad1
checksum-spare: Environment vars end
checksum-spare: ACTION to be taken: "degrade"
checksum-spare: Vdev /dev/sdab1 not found in pool test_pool 

the latter issue reported in the following script segment (dmesg is the local logging function - debug instrumentation):

```
 ...
        if [ -n "${ACTION}" ]; then

        # debug instrumentation
        dmesg "ACTION to be taken: \"$ACTION\""

        # Device is already FAULTED or DEGRADED                                                                                                                                                                                   
        set -- `vdev_status ${ZEVENT_POOL} ${ZEVENT_VDEV_PATH}`
        ZEVENT_VDEV_PATH_FOUND=$1
        STATUS=$2
        # debug instrumentation
        if [ -z $ZEVENT_VDEV_PATH_FOUND ]; then
            dmesg "Vdev ${ZEVENT_VDEV_PATH} not found in pool ${ZEVENT_POOL}"
            exit 10
        fi
        # debug instrumentation
        if [ -z $STATUS ]; then
            dmesg "Unexpected status \"$STATUS\" for vdev ${ZEVENT_VDEV_PATH} in pool ${ZEVENT_POOL}"
            exit 10
        fi
...
```

Checking vdev_status(), seems clear that it parses output of the zpool status command, and the latter uses device (as opposed to partition) names in its output:
...

```
NAME        STATE     READ WRITE CKSUM
test_pool   ONLINE       0     0     0
  mirror-0  ONLINE       0     0     0
    sdab    ONLINE       0     0 3.93K
    sdac    ONLINE       0     0     0
spares
  sdad      AVAIL   
```

... 
e.g. sdab is used, and not sdab1. That is why things do not quite work. Other scripts might have similar issues.

**Comment by mailinglists35** [comment]
I've pulling my hair for long time, looking first at #2449 without success until I have found your explanation, @bprotopopov 
Thank you!

This is really annoying!

My disks are partitioned like this:

```
root@voyage:/tmp# sgdisk -p /dev/sda
Disk /dev/sda: 976707632 sectors, 465.7 GiB
Logical sector size: 512 bytes
Disk identifier (GUID): 41B88685-4F31-4C2F-9978-5B6F294C6B7F
Partition table holds up to 128 entries
First usable sector is 34, last usable sector is 976707598
Partitions will be aligned on 2048-sector boundaries
Total free space is 2029 sectors (1014.5 KiB)

Number  Start (sector)    End (sector)  Size       Code  Name
   1            2048       976691199   465.7 GiB   BF01  zfs
   9       976691200       976707583   8.0 MiB     BF07
```

`zpool status` show my device names, `zdb -l devicename` shows no labels, `zdb -l partitionname` shows correct information for each `children` entry (`whole_disk: 1` and `path: partition name`)  however `io-spare.sh` script gets the *partition* name from `zed` and never finds a spare

**Comment by mailinglists35** [comment]
In my 0.6.4 debian io-spare.sh I have modified these lines in vdev_status().
It is wrong but it solved my autoreplace issue.

vdev_status original:

```
vdev_status() {
        local POOL=$1
        local VDEV=`basename $2`
        local T='       '       # tab character since '\t' isn't portable

        ${ZPOOL} status ${POOL} | sed -n -e \
            "s,^[ $T]*\(.*$VDEV\(-part[0-9]\+\)\?\)[ $T]*\([A-Z]\+\).*,\1 \3,p"
        return 0
}
```

modifications: strip `part1` (I only have zfs on part1) from `VDEV` and from output of `${ZPOOL} | sed`

modified:
```
vdev_status() {
        local POOL=$1
        local VDEV=`basename $2|sed 's/-part1$//g'`
        local T='       '       # tab character since '\t' isn't portable

        ${ZPOOL} status ${POOL} | sed -n -e \
            "s,^[ $T]*\(.*$VDEV\(-part[0-9]\+\)\?\)[ $T]*\([A-Z]\+\).*,\1 \3,p" \
            -e 's/-part1$//g'
        return 0
}
```

**Comment by behlendorf** [comment]
Releated to #4673 FMA work.

**Comment by behlendorf** [comment]
This functionality has been implemented in 976246fadde25790cec3de50c01c689d98d84e0c.

**Comment by mailinglists35** [comment]
awesome! will this be included in 0.7.0 release?

**Comment by behlendorf** [comment]
Yes
